src/test/java/com/google/step/servlets/FollowServletTest.java

```diff
@@ -0,0 +1,129 @@
+package com.google.step.servlets;
+
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.step.datamanager.FollowManager;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class FollowServletTest {
+
+  private static final long ID = 123;
+
+  private static final String RESTAURANT_PATH = "/restaurants/" + ID;
+  private static final String TAGS_PATH = "/tags/" + ID;
+  private static final String USERS_PATH = "/users/" + ID;
+
+  private static final String PATH_MISSING_ID = "/";
+  private static final String PATH_INVALID_NAME = "/trash/123";
+  private static final String PATH_INVALID_ID = "/tags/123/trash";
+
+  private FollowServlet servlet;
+  private FollowManager followManager;
+
+  @Before
+  public void setUp() {
+    followManager = mock(FollowManager.class);
+    servlet = new FollowServlet(followManager);
+  }
+
+  @Test
+  public void testDoPost_restaurant_sucess() throws IOException {
+    HttpServletRequest request = mock(HttpServletRequest.class);
+    HttpServletResponse response = mock(HttpServletResponse.class);
+
+    when(request.getPathInfo()).thenReturn(RESTAURANT_PATH);
+
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    when(response.getWriter()).thenReturn(writer);
+
+    servlet.doPost(request, response);
+
+    verify(response, never()).setStatus(HttpServletResponse.SC_BAD_REQUEST);
+    verify(followManager).followRestaurant(anyLong(), eq(ID));
+  }
+
+  @Test
+  public void testDoPost_tag_sucess() throws IOException {
+    HttpServletRequest request = mock(HttpServletRequest.class);
+    HttpServletResponse response = mock(HttpServletResponse.class);
+
+    when(request.getPathInfo()).thenReturn(TAGS_PATH);
```

Comment:
Since these constants are only used once, I'd prefer not making variables out of them. Rather just use their values here directly. Same for other tests.
```suggestion
    when(request.getPathInfo()).thenReturn("/tags/" + ID);
```